src/parser/jobs/rpr/modules/Reaver.tsx

```diff
@@ -0,0 +1,137 @@
+import {Plural, Trans} from '@lingui/react'
+import {DataLink} from 'components/ui/DbLink'
+import {Event, Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {filter, oneOf} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {Data} from 'parser/core/modules/Data'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+const SEVERITY_TIERS = {
+	1: SEVERITY.MEDIUM,
+	2: SEVERITY.MAJOR,
+}
+
+const SOUL_GAIN = 1 //Soul moves grant 1 reaver
+const GLUTTONY_GAIN = 2 //Gluttony grants 2 reavers, also therotically max
+
+export class Reaver extends Analyser {
+	static override handle = 'reaver'
+
+	@dependency private data!: Data
+	@dependency private suggestions!: Suggestions
+
+	//Trackers
+	private haveReaver = false
+
+	private droppedReaver = 0
+	private currentReaverStacks = 0
+
+	//groups
+	private TheMakers = [
+		this.data.actions.BLOOD_STALK.id,
+		this.data.actions.GRIM_SWATHE.id,
+		this.data.actions.UNVEILED_GALLOWS.id,
+		this.data.actions.UNVEILED_GIBBET.id,
+		this.data.actions.GLUTTONY.id,
+	]
+
+	private TheGoodOnes = [
+		this.data.actions.GUILLOTINE.id,
+		this.data.actions.GIBBET.id,
+		this.data.actions.GALLOWS.id,
+	]
+
+	private TheBadsOnes = [ //Generators handle elsewhere
+		this.data.actions.SLICE.id,
+		this.data.actions.WAXING_SLICE.id,
+		this.data.actions.INFERNAL_SLICE.id,
+		this.data.actions.HARVEST_MOON.id,
+		this.data.actions.SOULSOW.id, //Yes, even this takes it.
+		this.data.actions.HARPE.id,
+		this.data.actions.SHADOW_OF_DEATH.id,
+		this.data.actions.WHORL_OF_DEATH.id,
+		this.data.actions.SOUL_SCYTHE.id,
+		this.data.actions.SOUL_SLICE.id,
+		this.data.actions.SPINNING_SCYTHE.id,
+		this.data.actions.NIGHTMARE_SCYTHE.id,
+		this.data.actions.PLENTIFUL_HARVEST.id,
+	]
```

Comment:
Prefer these as consts and then in your hooks, use `this.data.matchActionId` to fill the IDs instead of `oneOf` in your action filter. Generally, we don't use leading capitals in names either, but with a const it's all caps so whatever.

Reply:
Moved to Consts and tried to make names clearer with Stack_makers, Stack_droppers and Stack_Skills_To_Use, not fully sold on my brain's naming for the 3rd one while trying to keep Stack as the first word.